mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/error.rs

```diff
@@ -29,6 +29,9 @@ pub enum RingBufferError {
     #[error("Buffer is full and messages must be drained to continue")]
     Full,
 
+    #[error("But has insufficient space to insert data: required: {required}b, but only {free}b available")]
```

Comment:
typo? 0_o

Reply:
What's the difference b/w `Full` and `InsufficientSpace`? It seems like we can have just one `InsufficientSpace` that covers everything.

Reply:
That is a fair point, I would prefer the InsufficientSpace over Full :)

Reply:
there is a special flag for 'Full' which made me add additional error variant.

Reply:
Yeah, the `Full` (and the current error in pipeline logs) was also useful for ignoring if we are full. But we should be able to switch to the `InsufficientSpace` version everywhere and get much more useful messages in logs